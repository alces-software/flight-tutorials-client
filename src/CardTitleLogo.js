// @flow
/*=============================================================================
 * Copyright (C) 2017 Stephen F. Norledge and Alces Software Ltd.
 *
 * This file is part of Flight Launch.
 *
 * All rights reserved, see LICENSE.txt.
 *===========================================================================*/
import React from 'react';

class CardTitleLogo extends React.Component {
  props: {
    logoUrl?: string,
  }

  state = {
    showImage: false,
  }

  showImage = () => {
    this.setState({ showImage: true });
  }

  render() {
    const { logoUrl } = this.props;

    const style = {
      display: this.state.showImage ? 'initial' : 'none',
    };

    return (
      logoUrl ? <img
        className="card-title-logo"
        alt="Logo for card"
        src={logoUrl}
        style={style}
        onLoad={this.showImage}
      />
      : null
    );
  }
}

export default CardTitleLogo;
