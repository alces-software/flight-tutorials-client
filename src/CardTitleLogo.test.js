// @flow
/*=============================================================================
 * Copyright (C) 2017 Stephen F. Norledge and Alces Flight Ltd.
 *
 * This file is part of Flight Tutorials.
 *
 * All rights reserved, see LICENSE.txt.
 *===========================================================================*/
import React from 'react';
import { render } from 'react-dom';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import CardTitleLogo from './CardTitleLogo';

it('renders without crashing', () => {
  const node = document.createElement('div');
  render(<CardTitleLogo logoUrl="http://localhost/my/logo.png" />, node);
});

it('renders correctly', () => {
  const tree = renderer.create(
    <CardTitleLogo logoUrl="http://localhost/my/logo.png" />,
  ).toJSON();

  expect(tree).toMatchSnapshot();
});

it('renders correctly if not given a logoUrl', () => {
  const tree = renderer.create(
    <CardTitleLogo />,
  ).toJSON();

  expect(tree).toMatchSnapshot();
});

it('hides the logo initially', () => {
  const wrapper = shallow(
    <CardTitleLogo logoUrl="http://localhost/my/logo.png" />,
  );

  expect(wrapper).toHaveStyle('display', 'none');
});

it('shows the logo once it has loaded', () => {
  const wrapper = shallow(
    <CardTitleLogo logoUrl="http://localhost/my/logo.png" />,
  );

  wrapper.prop('onLoad')();

  expect(wrapper).toHaveStyle('display', 'initial');
});
