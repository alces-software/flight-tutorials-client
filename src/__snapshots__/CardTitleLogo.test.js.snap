// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<img
  alt="Logo for card"
  className="card-title-logo"
  onLoad={[Function]}
  src="http://localhost/my/logo.png"
  style={
    Object {
      "display": "none",
    }
  }
/>
`;

exports[`renders correctly if not given a logoUrl 1`] = `null`;
